Tweak Development Setup Instructions

jtigger opened the pull request:
- be explicit about versions of dependencies.
- rely on package managers for installing software.
- clarify use of Docker/Vagrant
- switch use the imperative consistently in the section titles as these are instructions.
- clarify OAuth configuration steps.

Motardo commented on docs/setting-up-local-development.md:
```diff
@@ -2,73 +2,187 @@
+- [Install the prerequisites](#install-the-prerequisites)
+  - [Running exercism.io directly on your computer](#running-exercism-io-directly-on-your-computer)
+  - [Running exercism.io in a Docker container via Vagrant](#running-exercism-io-in-a-docker-container-via-vagrant)
+- [Initialize exercism.io](#initialize-exercism-io)
```
The links `#initialize-exercism-io`, `#run-exercism-io-locally`, and `#configure` (which I presume should link to `Configure Your Local Copy of exercism.io` don't seem to be working. I'm not sure but markdown might be having trouble generating the links because of the period in exercism.io

Motardo commented:
@jtigger I think that changing exercism-io to exercismio should fix the links:

```vim
s/#initialize-exercism-io/#initialize-exercismio/
s/#run-exercism-io-locally/#run-exercismio-locally/
s/#configure/#configure-your-local-copy-of-exercismio/
```

jtigger commented:
Thanks for the review, @Motardo.  Sorry for the long delay in response.

I've fixed those links (and updated the TOC a little too).

I appreciate the fact that it wasn't an Exercism end-user who reported these defects.  Thank you!

jtigger commented on docs/setting-up-local-development.md:
```diff
@@ -2,73 +2,187 @@
+- For other systems, see the [Node.js docs](https://github.com/joyent/node/wiki/Installing-Node.js-via-package-manager).
 
-_Again this is assuming you are forwarding port 3000 to 3030 in your Vagrantfile, adjust accordingly to your environment_
+With this, we're ready to download, build and run exercism.io locally (skip down to [The Code](#get-the-source-code))...
```
oops!  Missed this one... on it...

jtigger commented:
Got that "last" broken link. :)

kytrinyx commented:
Thank you, @jtigger!
